lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart

```diff
@@ -985,30 +938,45 @@ ViewListDiffResult? diffViewList(List<int> active, List<int> next) {
   if (active.isEmpty || next.isEmpty) {
     return null;
   }
-  // If the [active] and [next] lists are in the expected form described above,
-  // then either the first or last element of [next] will be in [active].
-  int index = active.indexOf(next.first);
-  if (index != -1) {
-    // Verify that the active composition order is contained, in order, in the
-    // next composition order.
+
+  // This is tried if the first element of the next list is contained in the
+  // active list at `index`. If the active and next lists are in the expected
+  // form, then we should be able to iterate from `index` to the end of the
+  // active list where every element matches in the next list.
+  ViewListDiffResult? lookForwards(int index) {
```

Comment:
Is this `lookForwards` function being defined inside the body of the `diffViewList` method, or is the formatting doing something weird?

Shouldn't these parameters passed in to `lookForwards` and `lookBackwards` instead, so we don't "bloat" the closure?

Maybe `diffViewList`, `lookForwards` and `lookBackwards` can be extracted to a separate `embedded_views_diff.dart` (with `ViewListDiffResult`?)?

Reply:
I moved the diffing code to a new file but kept the helper functions as internal. I think it's a little clearer that way because they will never be used outside of `diffViewList` and the closure bloat is minimal.